### DISCUSSION TITLE
Which framework for the website

### DISCUSSION THREAD

**Discussion by sidarth-23** [vim/website_next_generation#4] [Ideas]
In my opinion, angular seems fit as it is relatively stable with many first party libraries making maintenance easy. I think it would be a better choice to decide upon as a community before putting in our efforts into it. 

Let's discuss on it

**Comment by manuel-pross** [comment]
Thanks for your input Sidarth!

You are right when you say that Angular, due to its nature as a framework, as opposed to React as a library, has the advantage that many functionalities are provided by Angular without relying on third party libraries. As you correctly pointed out, this could make maintenance easier.

However, in this case we think that the maintenance of the Vim website, which does not have a complicated UI with many user interactions, is not the primary problem and therefore not the focus. In short, the codebase of the website will not be particularly complex.

We had a long discussion internally and decided to use Next.js for the following reasons:

1. We can have a not insignificant part of the content generated statically with the website bundle. In these cases it is not necessary to ship a large JS bundle to the browser. Next JS integrates this possibility perfectly.
2. We as a team started web development in the time when React as a frontend technology had already overtaken Angular in terms of developer satisfaction (current status: https://2022.stateofjs.com/en-US/libraries/front-end-frameworks/). Furthermore, we know React better and are confident that we can deliver a better result with it. The advantage you mentioned, that Angular already has its own solution for most use cases, becomes a disadvantage here in our opinion, due to it's larger overhead and because it takes more time to learn it.
3. As mentioned in point two, React is currently more popular than Angular. This has the advantage for this open source project that more people are able to contribute to it.

I've already heard of Angular Universal, which like Next.js also offers the possibility of SSR. However, we all haven't used that yet, so I refer to point two and three here.

**Reply by sidarth-23** [reply]
Sure then. I'll put in my best possible effort to make the transition possible. It will mean a new learning curve for me, but I'll take the challenge anyways. Hope my contributions are useful for the team

**Comment by oxdef** [comment]
Hey! :wave:
 
I suggest to have project landing and community areas (scripts, user account) be separated as well as making the landing  (including main page, News,  About Vim, download sections, etc.) with static generated site approach (already [mentioned above](https://github.com/vim/website_next_generation/discussions/4#discussioncomment-7379381)). Such way has following advantages:

- It is secure and lightweight by default (no back-end needed)
- GitHub pages could be utilized for hosting, including TLS certificates (so no spending money on hosting)
- It will provide more flexibility

It is pretty popular approach for many free and open-source projects.
